samples/StaticFilesAuth/Startup.cs

```diff
@@ -98,12 +110,45 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IAuthori
 
             app.Map("/MapAuthenticatedFiles", branch =>
             {
-                MapAuthenticatedFiles(branch, files);
+                // Blanket authorization, any authenticated user is allowed access to these resources.
+                branch.UseAuthorization(new AuthorizationOptions
```

Comment:
I don't really like creating a stand alone AuthorizationOptions which is detached from the normal AuthorizationOptions that someone might have configured normally.

What if this was `branch.UseAuthorization("AuthenticatedFilesPolicy")` which has the middleware automatically combine the policy specified before doing Authorize, but doesn't use a decoupled options instance.

So each instance of the middleware sorta behaves like an Authorize attribute, where you can add additional requirements.